### ISSUE TITLE
-Wunneeded-internal-declaration in drivers/bluetooth/{hci_ll|btusb}.c

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#207]
```
drivers/bluetooth/hci_ll.c:759:34: warning: variable 'hci_ti_of_match' is not needed and will not be emitted [-Wunneeded-internal-declaration]
static const struct of_device_id hci_ti_of_match[] = {
                                 ^
```

**Comment by nathanchance** [comment]
Ugh more `MODULE_DEVICE_TABLE` stuff...

I'm assuming that `CONFIG_OF` isn't enabled in your config so the only thing anchoring `hci_ti_of_match` and `btusb_match_table` to the file is `MODULE_DEVICE_TABLE` and `of_match_ptr`, the latter of which just returns NULL when `CONFIG_OF` isn't enabled.

I guess this, #203, and #205   should be fixed with `__used` like #169 and #178 but I'm wondering if there is a way to fix this in one place for the future.

**Comment by nickdesaulniers** [comment]
dupe of #232
